### DISCUSSION TITLE
software/2022/05/02/loongarch64-toolchain/

### DISCUSSION THREAD

**Discussion by giscus** [jiegec/blog-source#11] [General]
# software/2022/05/02/loongarch64-toolchain/

杰哥的{运维，编程，调板子}小笔记

https://jia.je/software/2022/05/02/loongarch64-toolchain/

**Comment by hurryboylqs** [comment]
你好，我在ubuntu24 上用https://github.com/loongson/build-tools?tab=readme-ov-file
这里提供的gcc交叉编译工具，编译EDK2 BaseTools的时候报错：
loongarch64-unknown-linux-gnu-gcc crs ../libs/libCommon.a BasePeCoff.o BinderFuncs.o CommonLib.o Crc32.o Decompress.o EfiCompress.o EfiUtilityMsgs.o FirmwareVolumeBuffer.o FvLib.o MemoryFile.o MyAlloc.o OsPath.o ParseGuidedSectionTools.o ParseInf.o PeCoffLoaderEx.o SimpleFileParsing.o StringFuncs.o TianoCompress.o
/opt/cross-tools/bin/../lib/gcc/loongarch64-unknown-linux-gnu/14.2.0/../../../../loongarch64-unknown-linux-gnu/bin/ld: cannot find crs: No such file or directory
/opt/cross-tools/bin/../lib/gcc/loongarch64-unknown-linux-gnu/14.2.0/../../../../loongarch64-unknown-linux-gnu/bin/ld: cannot find ../libs/libCommon.a: No such file or directory
collect2: error: ld returned 1 exit status
make[2]: *** [../Makefiles/footer.makefile:18：../libs/libCommon.a] 错误 1
make[2]: 离开目录“/home/user/works/edk2-2024/BaseTools/Source/C/Common”
make[1]: *** [GNUmakefile:79：Common] 错误 2
make[1]: 离开目录“/home/user/works/edk2-2024/BaseTools/Source/C”
make: *** [GNUmakefile:19：Source/C] 错误 2
make: 离开目录“/home/user/works/edk2-2024/BaseTools”


请问知道是什么原因吗？

**Reply by jiegec** [reply]
请在 https://github.com/loongson/build-tools 提 issue。

**Reply by jiegec** [reply]
可以参考 [AOSC-Dev/LoongArchQemuVirtFirmware](https://github.com/AOSC-Dev/LoongArchQemuVirtFirmware/blob/master/.github/workflows/build.yml) 的代码，它在 GitHub CI 上编译 EDK II 的 LoongArch firmware。
